tests/models/clvp/test_modeling_clvp.py

```diff
@@ -0,0 +1,651 @@
+        model = CLVPTextModelWithProjection(config=config)
+        model.to(torch_device)
+        model.eval()
+        with torch.no_grad():
+            result = model(input_ids, attention_mask=input_mask)
+            result = model(input_ids)
+        self.parent.assertEqual(result.last_hidden_state.shape, (self.batch_size, self.seq_length, self.hidden_size))
+        self.parent.assertEqual(result.text_embeds.shape, (self.batch_size, self.projection_dim))
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        config, input_ids, input_mask = config_and_inputs
+        inputs_dict = {"input_ids": input_ids, "attention_mask": input_mask}
+        return config, inputs_dict
+
+
+@require_torch
+class CLVPTextModelTest(ModelTesterMixin, unittest.TestCase):
+    all_model_classes = (CLVPTextModel, CLVPTextModelWithProjection) if is_torch_available() else ()
+    fx_compatible = False
+    test_pruning = False
+    test_head_masking = False
+    test_torchscript = False
+
+    def setUp(self):
+        self.model_tester = CLVPTextModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=CLVPTextConfig, hidden_size=64)
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    def test_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(*config_and_inputs)
+
+    def test_model_with_projection(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model_with_projection(*config_and_inputs)
+
+    @unittest.skip(reason="CLVPTextModelWithProjection does not output loss")
+    def test_training(self):
+        pass
+
+    @unittest.skip(reason="CLVPTextModelWithProjection does not output loss")
+    def test_training_gradient_checkpointing(self):
+        pass
+
+    @unittest.skip(reason="CLVP does not use inputs_embeds")
+    def test_inputs_embeds(self):
+        pass
+
+    @unittest.skip(reason="CLVPTextModel has no base class and is not available in MODEL_MAPPING")
+    def test_save_load_fast_init_from_base(self):
+        pass
+
+    @unittest.skip(reason="CLVPTextModel has no base class and is not available in MODEL_MAPPING")
+    def test_save_load_fast_init_to_base(self):
+        pass
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in CLVP_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = CLVPTextModel.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+    @slow
+    def test_model_with_projection_from_pretrained(self):
+        for model_name in CLVP_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = CLVPTextModelWithProjection.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+            self.assertTrue(hasattr(model, "text_projection"))
```

Comment:
Is there a particular reason why some of the tests are not the same, since the code is basically the same except for variable names?
If not, I think you might want to test `CLVPTransformer` and `CLVPTransformerWithProjection`, if you indeed change `CLVPTextModel|CLVPSpeechModel -> CLVPTransformer`.

Reply:
we now only test `CLVPTransformerWithProjection` and `CLVPModel`